polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -1476,3 +1477,423 @@ ValueCategory MLIRScanner::VisitCallExpr(clang::CallExpr *expr) {
   return CallHelper(ToCall, objType, args, expr->getType(),
                     expr->isLValue() || expr->isXValue(), expr, callee);
 }
+
+std::pair<ValueCategory, bool>
+MLIRScanner::EmitGPUCallExpr(clang::CallExpr *expr) {
```

Comment:
Moved from clang-mlir.cc because it is called from `VisitCallExpr` which is in this file.